### PR TITLE
feat: adding database models

### PR DESCRIPTION
**Pull request by natilou**
- User
- Metric
- Challenge
- Tournament
- MetricDailyHistory

### TODO:
- create admin user, maybe with [sqladmin](https://aminalaee.dev/sqladmin/)
- configure `mysql` engine with env viriable from docker compose, now we are using `sqlite`

### Documentation:
- SQLAlchemy `mapped_column`: https://docs.sqlalchemy.org/en/20/orm/mapping_api.html#sqlalchemy.orm.mapped_column
-  SQLAlchemy `Mapped`:  https://docs.sqlalchemy.org/en/20/orm/internals.html#sqlalchemy.orm.Mapped
- SQLAlchemy Many to Many relationship: https://docs.sqlalchemy.org/en/20/orm/basic_relationships.html#many-to-many

### CODE REVIEW

File: models.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
+from datetime import date
+
+from sqlalchemy import CheckConstraint, Column
+from sqlalchemy import Enum as SqlAlchemyEnum
+from sqlalchemy import ForeignKey, Table
+from sqlalchemy.orm import DeclarativeBase, Mapped, mapped_column, relationship
```
Comment (matiasbertani) [inline models.py:6]: Could be better to wraps the imports (here and the some imports avobe) between brackets??

```python
from sqlalchemy.orm import (
    DeclarativeBase,
    Mapped,
    mapped_column,
    relationship,
)
```

File: models.py [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+
+association_user_metric_history = Table(
+    "association_user_metric_history",
+    Base.metadata,
+    Column("metric_daily_history_id", ForeignKey("metrics_daily_history.id")),
+    Column("user_id", ForeignKey("users.id")),
+)
+
+association_metric_metric_history = Table(
```
Comment (matiasbertani) [inline models.py]: (more specifically at this association table, but at the other ones too)
What do you think of using `and` or `with` to separate the entities we are relating/associating? could we help us to read the association better? 
We will end with something like this `association_metric_and_metric_history` or `association_metric_with_metric_history`. 

Or this probably lead us to very long names with non-purpose?

**Review by guillermoojeda — APPROVED** [review]

**Review by matiasbertani — APPROVED** [review]
Looks all good to me, I made a few comments, but nothing to block the merge. Another thing that could be great for the future is to create a directory to leave all the database-related modules.

File: models.py (hunk 1 above)
Comment (natilou) [reply models.py:6]: I did what you suggest, but `isort` hook changed to this way. For the next PR I will check if we have to change some of the `isort` configuration.

File: models.py (hunk 2 above)
Comment (natilou) [reply models.py]: Yeah, I wasn't sure about the name of the associations, specially the one that you mention. I think that, in this case, using `and` or `with` would a better approach, despite the long name. I will make the change right now

File: models.py (hunk 1 above)
Comment (matiasbertani) [reply models.py:6]: Excellent

**Pull request merged**